plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeSecurityConfig.java

```diff
@@ -18,29 +18,21 @@
 
 import javax.validation.constraints.NotNull;
 
-import static io.trino.plugin.deltalake.DeltaLakeSecurityConfig.DeltaLakeSecurity.ALLOW_ALL;
+import static io.trino.plugin.deltalake.DeltaLakeSecurityModule.ALLOW_ALL;
 
 public class DeltaLakeSecurityConfig
 {
-    public enum DeltaLakeSecurity
-    {
-        ALLOW_ALL,
-        READ_ONLY,
-        SYSTEM,
-        FILE,
-    }
-
-    private DeltaLakeSecurity securitySystem = ALLOW_ALL;
+    private String securitySystem = ALLOW_ALL;
 
     @NotNull
-    public DeltaLakeSecurity getSecuritySystem()
+    public String getSecuritySystem()
     {
         return securitySystem;
     }
 
     @Config("delta.security")
     @ConfigDescription("Authorization checks for Delta Lake connector")
-    public DeltaLakeSecurityConfig setSecuritySystem(DeltaLakeSecurity securitySystem)
+    public DeltaLakeSecurityConfig setSecuritySystem(String securitySystem)
```

Comment:
This has the drawback that there is no validation for the provided value.


I used successfully for testing:

```
delta.security=foobar
```

In `io.trino.plugin.deltalake.DeltaLakeSecurityModule#bindSecurityModule` `foobar` goes through because no security module is bound.


Please add an (eventually extensible) validation mechanism for the value provided for the access control.
Once we get to an acceptable solution, please keep in mind that a similar mechanism may need to be applied for hive as well (follow-up PR).

Reply:
How it works with `hive.security`? Is this a preexisting issue?

Maybe we could validate, once guice is initialized that expected beans are there. WDYT? The issue is that for `system` security mode we don't bind an access control.

Reply:
Setting `hive.security` to `foobar` is also possible, but there is a side-effect of doing this which stops the hive connector to successfully bootstrap:

```
1) [Guice/JitDisabled]: Explicit bindings are required and AccessControlMetadataFactory is not explicitly bound.
  at HiveMetadataFactory.<init>(HiveMetadataFactory.java:103)
      \_ for 17th parameter accessControlMetadataFactory
  at HiveModule.configure(HiveModule.java:97)
```

> Maybe we could validate, once guice is initialized that expected beans are there. WDYT?



I followed your suggestion and added in `HiveSecurityModule` a draft solution:

```
        bindSecurityModule(SQL_STANDARD, new SqlStandardSecurityModule());
        bindSecurityModule(ALLOW_ALL, new AllowAllSecurityModule());
        bindSecurityModule(SYSTEM, new SystemSecurityModule());

        binder.bind(AccessControlMetadataFactoryValidator.class).asEagerSingleton();
    ...

    private static class AccessControlMetadataFactoryValidator
    {
        @Inject
        public AccessControlMetadataFactoryValidator(AccessControlMetadataFactory accessControlMetadataFactory)
        {
            requireNonNull(accessControlMetadataFactory, "accessControlMetadataFactory is null");
        }
    }
```

Reply:
I was curious to see how this could be done and came up with a better solution in `HiveSecurityModule`

```
        install(ConditionalExceptionModule.conditionalExceptionModule(
                SecurityConfig.class,
                securityConfig -> !isSecuritySystemValid(securityConfig),
                securityConfig -> new ApplicationConfigurationException(
                        ImmutableList.of(new Message("Error: Configuration property 'hive.security' value '%s' is invalid".formatted(securityConfig.getSecuritySystem()))),
                        ImmutableList.of())));

...
    // This could be added into an extension point
    private boolean isSecuritySystemValid(SecurityConfig securityConfig){
        return Set.of(LEGACY, FILE, READ_ONLY, SYSTEM).contains(securityConfig.getSecuritySystem());
    }

...

    // This could be added eventually to airlift  (it is inspired from `io.airlift.configuration.ConditionalModule`)
    private static class ConditionalExceptionModule<T>
            extends AbstractConfigurationAwareModule
    {
        private final Class<T> config;
        private final Predicate<T> predicate;
        private final Function<T, RuntimeException> exceptionFunction;

        public static <T> ConditionalExceptionModule conditionalExceptionModule(Class<T> config, Predicate<T> predicate, Function<T, RuntimeException> exceptionFunction)
        {
            return new ConditionalExceptionModule(config, predicate, exceptionFunction);
        }

        private ConditionalExceptionModule(Class<T> config, Predicate<T> predicate, Function<T, RuntimeException> exceptionFunction)
        {
            this.config = requireNonNull(config, "config is null");
            this.predicate = requireNonNull(predicate, "predicate is null");
            this.exceptionFunction = requireNonNull(exceptionFunction, "exceptionFunction is null");
        }

        protected void setup(Binder binder)
        {
            T configuration = this.buildConfigObject(this.config);
            if (this.predicate.test(configuration)) {
                throw exceptionFunction.apply(configuration);
            }
        }
    }
```

While bootstrapping the hive connector we can see now:

```
Caused by: io.airlift.bootstrap.ApplicationConfigurationException: Configuration errors:

1) Error: Error: Configuration property 'hive.security' value 'foobar' is invalid

1 error
	at io.trino.plugin.hive.security.HiveSecurityModule$ConditionalErrorModule.setup(HiveSecurityModule.java:123)
	at io.airlift.configuration.AbstractConfigurationAwareModule.configure(AbstractConfigurationAwareModule.java:46)
	at com.google.inject.spi.Elements$RecordingBinder.install(Elements.java:409)
	at io.airlift.configuration.AbstractConfigurationAwareModule.install(AbstractConfigurationAwareModule.java:70)
	at io.trino.plugin.hive.security.HiveSecurityModule.setup(HiveSecurityModule.java:71)
```

cc @dain

Reply:
I think this is a separate issue. Can you please file an issue for this and post a PR with your idea?

Reply:
Nevermind. I have attempted to come up with a solution for this issue, but it don't find anything.

Probably adding a validator class as sugested in https://github.com/trinodb/trino/pull/14153#discussion_r976038437 is a good way to go. Unfortunately the user still has to dig up to find out that the `delta.security` setting is incorrect.

Reply:
> Probably adding a validator class as sugested in #14153 (comment) is a good way to go. Unfortunately the user still has to dig up to find out that the delta.security setting is incorrect.

This exact approach doesn't work for Delta as we by default bind the interface to SYSTEM. I did implement something similar that checks if we have been supplied a non-SYSTEM value and still get the SYSTEM interface implementation.

Reply:
Removed the validation as it is an existing shortcoming and out of scope in this PR as disussed with @kokosing